Libraries/LibJS/Runtime/ProxyObject.cpp

```diff
@@ -466,9 +467,79 @@ Value ProxyObject::delete_property(PropertyName name)
 
 void ProxyObject::visit_children(Cell::Visitor& visitor)
 {
-    Object::visit_children(visitor);
+    Function::visit_children(visitor);
     visitor.visit(&m_target);
     visitor.visit(&m_handler);
 }
 
+Value ProxyObject::call(Interpreter& interpreter) {
+    if (!is_function())
+        return interpreter.throw_exception<TypeError>(ErrorType::NotAFunction, Value(this).to_string_without_side_effects().characters());
+
+    if (m_is_revoked) {
+        interpreter.throw_exception<TypeError>(ErrorType::ProxyRevoked);
+        return {};
+    }
+    auto trap = m_handler.get("apply");
+    if (interpreter.exception())
+        return {};
+    if (trap.is_empty() || trap.is_undefined() || trap.is_null())
+        return static_cast<Function&>(m_target).call(interpreter);
+    if (!trap.is_function())
+        return interpreter.throw_exception<TypeError>(ErrorType::ProxyInvalidTrap, "apply");
+
+    MarkedValueList arguments(interpreter.heap());
+    arguments.values().append(Value(&m_target));
+    arguments.values().append(Value(&m_handler));
```

Comment:
Can we just use `MarkedValueList::append()`?

```suggestion
    arguments.append(&m_target);
    arguments.append(&m_handler);
```

(+Same suggestion where it applies elsewhere)

Reply:
Huh, for some reason I've done this in all of my LibJS code. I think I copied it from somewhere I saw (that no longer does it) and just never bothered to actually look at that class to see that method. I think I'll leave it here, and fix all places where I've done it in a new QOL PR that will also fix some other issues.